include/deal.II/differentiation/sd/symengine_scalar_operations.h

```diff
@@ -146,8 +146,317 @@ namespace Differentiation
        */
       bool
       is_valid_substitution_symbol(const SymEngine::Basic &entry);
+
+      /**
+       * A convenience function to set the @p value associated with
+       * the @p symbol in the @p substitution_map.
+       *
+       * Using this function ensures that the @p symbol is one that is
+       * valid specifically for the purpose of symbolic substitution.
+       * It must therefore represent a symbol or symbolic derivative,
+       * otherwise an error will be thrown.
+       */
+      void
+      set_value_in_symbol_map(
+        types::substitution_map &                     substitution_map,
+        const SymEngine::RCP<const SymEngine::Basic> &symbol,
+        const SymEngine::RCP<const SymEngine::Basic> &value);
```

Comment:
We pretty widely put the output argument (i.e., the one being modified) into the last slot. Any particular reason not to use this convention here?

Reply:
Yeah, this matches the arrangement used in the function in the non-internal namespace. I wanted to retain consistency in case someone chooses to open up the namespaces and use all the functions without the namespace qualification. To me it makes sense to retain this consistency. As a response to one of your other comments I'll explain why I chose this ordering.